src/webview/MessageList.js

```diff
@@ -143,79 +144,85 @@ class MessageListInner extends React.Component<Props> {
   // themself suddenly scrolled back to the bottom.
   //
   // So:
-  //  * We let this component render just once, and define
-  //    `shouldComponentUpdate` so as to prevent re-renders.
-  //    (Though there's no guarantee that it accomplishes that completely;
-  //    see docs/architecture/react.md .)
+  //  * We compute the HTML document just once and then always re-use that
+  //    initial value, using `useComputeOnce`.
   //  * When the props change, we compute a set of events describing the
   //    changes, and send them to our code inside the webview to execute.
   //
   // See also docs/architecture/react.md .
 
-  webviewRef = React.createRef<React$ElementRef<typeof WebView>>();
-  sendInboundEventsIsReady: boolean = false;
-  unsentInboundEvents: WebViewInboundEvent[] = [];
+  const webviewRef = React.useRef<React$ElementRef<typeof WebView>>();
```

Comment:
Hmm, I think both the [current React doc](https://reactjs.org/docs/hooks-reference.html#useref) and the nice new [beta doc](https://beta.reactjs.org/apis/react/useRef) for `useRef` aren't perfectly clear about what happens when you don't pass an `initialValue` argument to `useRef`.

If it means `.current` begins its life as `undefined`, there could theoretically be a problem when it runs into the `!== null` check in `sendInboundEvents`.

I would explicitly pass `null`, as we do in `useUncontrolledInput`, for example:

```js
const ref = useRef<React$ElementRef<typeof TextInput> | null>(null);
```

Reply:
Possibly `React.createRef()` (what we used before this commit) initialized `.current` to `null`. I say that because of the one word "back" in [the doc](https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element):

> React will assign the `current` property with the DOM element when the component mounts, and assign it back to `null` when it unmounts